### PR TITLE
Expand docker-compose.yml

### PR DESCRIPTION
**Pull request by soulshake**

### CODE REVIEW

File: script/docker-build-and-push [hunk 1]
```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+
+set -ev
+app_name="hub"
```
Comment (bnferguson) [inline script/docker-build-and-push]: Only change I'd make is adding something like https://github.com/travis-ci/travis-logs/blob/master/script/docker-build-and-push#L5-L8 here so it doesn't always build. We could also make it a conditional build step. Either way!

Comment (soulshake) [reply script/docker-build-and-push]: I modified as you suggested, and also added an external build trigger on Docker hub. (We should discuss this at some point)

Comment (soulshake) [reply script/docker-build-and-push]: Oh, I forgot about the private gem issue. Need to fix that still.

Comment (bnferguson) [reply script/docker-build-and-push]: > I modified as you suggested, and also added an external build trigger on Docker hub. (We should discuss this at some point)

Dope! And indeed.

**Pull request merged**